synapse/types/handlers/sliding_sync.py

```diff
@@ -69,15 +69,7 @@ class SlidingSyncConfig(SlidingSyncBody):
 
     user: UserID
     requester: Requester
-
-    # Pydantic config
-    class Config:
-        # By default, ignore fields that we don't recognise.
-        extra = Extra.ignore
-        # By default, don't allow fields to be reassigned after parsing.
-        allow_mutation = False
-        # Allow custom types like `UserID` to be used in the model
-        arbitrary_types_allowed = True
+    model_config = ConfigDict(extra="ignore", frozen=True, arbitrary_types_allowed=True)
```

Comment:
I'd like to keep the comments for each of these fields as they provide useful information.

```suggestion
    model_config = ConfigDict(
        # By default, ignore fields that we don't recognise.
		extra="ignore",
		# By default, don't allow fields to be reassigned after parsing.
		frozen=True,
		# Allow custom types like `UserID` to be used in the model.
		arbitrary_types_allowed=True,
)
```